Fix 2017.x compatibility of the Font Asset Baker

joonjoonjoon opened the pull request.

joonjoonjoon committed:
Update TMPFontAssetBaker.cs

Updated to be compatible with 2017.x and onward.

joonjoonjoon committed:
Properly instantiate materials

Fixed error in previous fix to have 2017.x compatibility

DarrenTsung commented on Editor/TMPFontAssetBaker/TMPFontAssetBaker.cs:
```diff
@@ -86,10 +86,25 @@ public static class TMPFontAssetBaker {
 				Shader shader = Shader.Find("TMPro/Distance Field");
+				if(shader == null) 
+				{
+					Shader.Find("TextMeshPro/Distance Field"); // Update namespace for 2018
```
This doesn't assign the shader found to the `shader` variable

DarrenTsung commented on Editor/TMPFontAssetBaker/TMPFontAssetBaker.cs:
```diff
@@ -86,10 +86,25 @@ public static class TMPFontAssetBaker {
 			if (materialReferences == null || materialReferences.Length <= 0) {
 				// Create new Material and add it as Sub-Asset
 				Shader shader = Shader.Find("TMPro/Distance Field");
+				if(shader == null) 
```
Follow coding convention of parens on the same line please:
```
if (shader == null) {
     // .. stuff
}
```

DarrenTsung commented on Editor/TMPFontAssetBaker/TMPFontAssetBaker.cs:
```diff
@@ -86,10 +86,25 @@ public static class TMPFontAssetBaker {
+			catch(NullReferenceException exception)
+			{
+				materialReferences = null;
+				Debug.LogWarning("Exception caught: " + exception.Message);
```
Should we log a warning if this an expected case?

joonjoonjoon committed:
Make sure character set doesn't include duplicates

The normal TMPRo window takes care of removing duplicates when an arbitrary characterset is given.
Adding .Distinct() to the existing select statement takes care of that.
